sources/api/apiserver/src/server/controller.rs

```diff
@@ -449,13 +522,141 @@ pub(crate) fn get_metadata_for_all_data_keys<D: DataStore, S: AsRef<str>>(
     Ok(result)
 }
 
+// Parses and validates the settings and metadata in pending transaction and
+// returns the constraint check result containing approved settings and metadata to
+// commit to live transaction.
+// We will pass this function as argument to commit transaction function.
+fn datastore_transaction_check<D>(
+    datastore: &mut D,
+    committed: &Committed,
+) -> Result<ConstraintCheckResult>
+where
+    D: DataStore,
+{
+    // Get settings to commit from pending transaction
+    let settings_to_commit = datastore
+        .get_prefix("settings.", committed)
+        .context(error::DataStoreSnafu { op: "get_prefix" })?;
+
+    // Get metadata from pending transaction
+    let mut transaction_metadata = datastore
+        .get_metadata_prefix("settings.", committed, &None as &Option<&str>)
+        .context(error::DataStoreSnafu {
+            op: "get_metadata_prefix",
+        })?;
+
+    // Vector(metadata_key, key, value)
+    let mut metadata_to_commit: Vec<(Key, Key, String)> = Vec::new();
+
+    // Parse and validate all the metadata enteries from pending transaction
+    for (key, value) in transaction_metadata.iter_mut() {
+        for (metadata_key, metadata_value) in value {
+            // For now we are only processing the strength metadata from pending
+            // transaction to live
+            if metadata_key.name() != "strength" {
+                warn!(
+                    "Metadata key is {}, Skipping the commit as we only allow Strength metadata.",
+                    metadata_key.name()
+                );
+                continue;
+            }
+
+            // strength in pending transaction
+            let pending_strength: String =
+                deserialize_scalar::<_, ScalarError>(&metadata_value.clone())
+                    .with_context(|_| error::DeSerializeSnafu {})?;
+
+            let pending_strength: Strength = pending_strength
+                .parse::<Strength>()
+                .context(error::ParseStrengthSnafu)?;
```

Comment:
Is there a way to make this look more like the `let committed_strength = ...` assignment below? That could help readability.